src/command/dev/protocol/mod.rs

```diff
@@ -1,9 +1,27 @@
+pub use follower::*;
+pub use leader::*;
+pub(crate) use socket::*;
+pub use types::*;
+
 mod follower;
 mod leader;
 mod socket;
 mod types;
 
-pub use follower::*;
-pub use leader::*;
-pub(crate) use socket::*;
-pub use types::*;
+macro_rules! create_socket_name {
```

Comment:
I played around with this locally, and it seems like we could get away with something like this in place of a macro:
```
pub(crate) fn create_socket_name<'s>(raw_socket_name: &'s str) -> std::io::Result<Name<'s>> {
    if GenericFilePath::is_supported() {
        raw_socket_name.clone().to_fs_name::<GenericFilePath>()
    } else {
        raw_socket_name.clone().to_ns_name::<GenericNamespaced>()
    }
}
```

This would pass the buck on unwrapping or mapping the result to the calling function and solve for aaron's comments below. Passing the buck to the caller for errors is generally something we tend to do instead of panicking.

In the calling code, it's simple enough to do something like:
```
let socket_name = create_socket_name(&raw_socket_name as &str)?;
```
This works cleanly because rover maps any error into an `anyhow::Error` [here](https://github.com/apollographql/rover/blob/main/src/error/mod.rs#L154).